dev/CommandBarFlyout/CommandBarFlyout.cpp

```diff
@@ -252,26 +252,135 @@ winrt::Control CommandBarFlyout::CreatePresenter()
-        // We need to specifically target the CommandBar for the shadow, not the default node far
-        // above that.
-        if (winrt::IFlyoutPresenter2 presenter2 = presenter)
+        primaryCornerRadius = unbox_value<winrt::CornerRadius>(cornerRadiusResource);
+    }
+
+    if (winrt::IControl7 presenterControl7 = presenter)
+    {
+        // When >21H1, we'll need to manage the presenter's CornerRadius when the overflow is opened/
+        // closed. In order for drop shadows to render correctly, the element with the shadow needs to have
+        // the correct corner radius as well. Otherwise the shadow will render with sharp corners.
+        if (SharedHelpers::Is21H1OrHigher())
+        {
+            presenterControl7.CornerRadius(primaryCornerRadius);
+        }
+        else
         {
-            presenter2.IsDefaultShadowEnabled(false);
+            presenterControl7.CornerRadius({ 0 });
+        }
+    }
+
+    m_presenter.set(presenter);
+
+    m_commandBarOpenedRevoker = commandBar->Opened(winrt::auto_revoke, {
+        [this, primaryCornerRadius](auto const&, auto const&)
+        {
+            if (winrt::IFlyoutBase5 thisAsFlyoutBase5 = *this)
+            {
+                // If we open the CommandBar, then we should no longer be in a transient show mode -
+                // we now know that the user wants to interact with us.
+                thisAsFlyoutBase5.ShowMode(winrt::FlyoutShowMode::Standard);
+            }
+
+            if (SharedHelpers::Is21H1OrHigher() && m_presenter)
+            {
+                if (winrt::IControl7 presenterControl7 = m_presenter.get())
+                {
+                    if (auto commandBar = winrt::get_self<CommandBarFlyoutCommandBar>(m_commandBar.get()))
+                    {
+                        if (m_secondaryCommands.Size() > 0 || commandBar->IsOpen())
+                        {
+                            auto cornerRadiusConverter = winrt::make_self<CornerRadiusFilterConverter>();
+
+                            // CommandBarFlyout already changes its corner radius depending on if the overflow is open
+                            // and in which direction through visual states. Since we're managing the corner radius
+                            // on the presenter manually, we'll need to mimic these changes onto the presenter's corner radius.
+                            if (commandBar->IsExpandedUp())
+                            {
+                                presenterControl7.CornerRadius(cornerRadiusConverter->Convert(primaryCornerRadius, winrt::CornerRadiusFilterKind::Bottom));
+                            }
+                            else
+                            {
+                                presenterControl7.CornerRadius(cornerRadiusConverter->Convert(primaryCornerRadius, winrt::CornerRadiusFilterKind::Top));
+                            }
```

Comment:
Instead of doing this manually, can we add a binding between the command bar's CornerRadius property and the presenter's CornerRadius property?  Seems like it would make a lot of this simpler.  You can use the [SetBinding method on FrameworkElement](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.frameworkelement.setbinding?view=winrt-19041) to do that programmatically.